hpolib/abstract_benchmark.py

```diff
@@ -114,7 +120,23 @@ def wrapper(self, configuration: Union[np.ndarray, ConfigSpace.Configuration, Di
                                 f'was {type(configuration)}')
 
             self.configuration_space.check_configuration(config)
-            return foo(self, configuration, **kwargs)
+
+            # Check the fidelities, if any
+            fidelity = self.opt_fidelity.get_dictionary()
+            if kwargs:
```

Comment:
You can rely on the fidelities being handed over as, e.g. fidelities={"epochs"=3}. So you can check for the existence of this dictionary in kwargs

Reply:
Why dont we then add the fidelity parameter in the wrapper parameters?

`def wrapper(self, configuration: Union[np.ndarray, ConfigSpace.Configuration, Dict], fidelity_space: Union[ConfigSpace.Configuration, Dict], **kwargs)`

Reply:
I agree with Phillip in that the current interface is more user friendly while the 'fidelities=...' approach would be more suitable for cases where a large number of fidelities are possible, thus creating potential for name clashes with other keyword arguments.

The change would also interfere with all existing code since it necessitates a change in the function call signature, although I'm not certain how much such code exists and how concerned we should be about interface changes.

Reply:
I thought it would be much easier for us (also in terms of testing) to rely on a unified interface which always takes the same arguments, `configuration, [fidelities, seed]`. I do understand that this will require to create a wrapper function for any optimizer we are currently aware of, but the interface is much cleaner. I need to think about this a bit more.

Reply:
I see the merits to the approach. Take your time, this change won't require a lot of effort from my end, at the very least. I'd just like to point out that we will need more changes to achieve total uniformity, as some of the benchmarks' objective functions have arguments that don't quite fit the bill for being fidelity parameters either. (e.g. NASCifar10BaseBenchmark's objective function has a boolean argument called 'reset').

My recommendation to deal with such settings is to have a separate benchmark-specific config dictionary/collection class with a more unified interface, so the objective function calls really only do exactly what they're supposed to do - provide a set of configurations to evaluate upon. But this is, again, hugely dependent upon whether or not it's worth the effort in our case, since such config collections are, by design, beneficial when dealing with a large number of settings.